gammapy/data/metadata.py

```diff
@@ -74,17 +76,69 @@ def validate_time(cls, v):
         else:
             raise ValueError("Incorrect time input value.")
 
+    @validator("target_position")
+    def validate_position(cls, v):
+        if v is None:
+            return SkyCoord(np.nan, np.nan, unit="deg", frame="icrs")
+        elif isinstance(v, SkyCoord):
+            return v
+        else:
+            raise ValidationError(
+                f"Incorrect position. Expect SkyCoord got {type(v)} instead."
+            )
+
     @classmethod
     def from_gadf_header(cls, events_hdr):
```

Comment:
Should we not start to add the version ID of a format? In order to deal with format evolution...

Reply:
I suppose that in the long run we might have to introduce several such functions for different formats and subsequent version. I suppose we won't have to deal with a huge variety of formats.

But in general I would find it more convenient to have all I/O related functions in specific namespaces/classes. The factory method could find the correct format from the header itself.

In the meantime, we can rename this `ObservationMetaData.from_header(cls, events_hdr, format="gadf")`.